doc/md/guides/frameworks/sqlc-declarative.md

```diff
@@ -0,0 +1,155 @@
+process.
+
+In this guide we will show you how Atlas can be used with sqlc in a declarative way, filling the gaps and providing a
+complete solution for building applications with sqlc.
+
+We assume that you already have a project using sqlc, if you are new to sqlc or don't have a project yet,
+[check the getting started guide](https://docs.sqlc.dev/en/latest/tutorials/getting-started-postgresql.html).
+
+## Schema
+
+To generate code, sqlc depends on the database schema and the queries to be generated against it. We need both the
+schema and queries, to ensure sqlc knows what types and queries it needs to generate.
+
+As your application’s schema evolves, drift between the desired schema and database can cause many issues with sqlc,
+since the most common time you will detect the errors is during the query execution.
+
+In this guide, we will show how sqlc users can utilize Atlas’s declarative schema migration workflow to ensure their
+database schemas are always in sync with their desired state.
+
+## Desired state
+
+Since both Atlas and sqlc can accept a single SQL file to describe the desired state of the database schema, this file
+can be the source of truth (or desired state) for both tools, letting each tool handle its part of the job.
+
+This is a great example of the power of
+the [declarative concept](https://atlasgo.io/concepts/declarative-vs-versioned#declarative-migrations), where we declare
+what we expect and let the tooling figure out how to reach the desired state.
+
+## Migrating the initial schema
+
+If you followed the sqlc tutorial, you may end up with a `schema.sql` file that looks like this:
+
+```sql title="schema.sql"
+CREATE TABLE authors
+(
+    id   BIGSERIAL PRIMARY KEY,
+    name text NOT NULL,
+    bio  text
+);
+```
+
+For the purpose of this guide, we will assume that our application is backed by a PostgreSQL database running in a local
+Docker container. You can start such a container by running:
+
+```shell
+docker run --rm -d --name atlas-sqlc -p 5432:5432 -e POSTGRES_PASSWORD=pass -e POSTGRES_DB=sqlc postgres
+```
+
+:::info
+While in this example we are using Postgres, this is not a requirement by sqlc or Atlas, you can switch to any other
+database as long the `schema.sql` is compatible.
+:::
+
+The first thing we do is to initialize the database schema, this can be accomplished with a simple
+command:
+
+```shell
+atlas schema apply \
+--url "postgres://postgres:pass@localhost:5432/sqlc?sslmode=disable" \
+--dev-url "docker://postgres" \
+--to "file://schema.sql"
+```
+
+After executing the command, you should see the planned changes like the example below:
+
+```text {2-5}
+-- Planned Changes:
+-- Add new schema named "public"
+CREATE SCHEMA "public";
+-- Create "authors" table
+CREATE TABLE "public"."authors" ("id" bigserial NOT NULL, "name" text NOT NULL, "bio" text NULL, PRIMARY KEY ("id"));
```

Comment:
Awesome. When we do it for versioned migrations, we'll need to scope the connection to a specific schema and avoid having the schema qualifier in identifiers (e.g., public.users => users).